.github/workflows/ci.yml

```diff
@@ -0,0 +1,114 @@
+name: ci
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - '**'
+env:
+  COVERAGE_OPTION: ./node_modules/.bin/nyc
+  NODE_VERSION: 10
+  PARSE_SERVER_TEST_TIMEOUT: 20000
+jobs:
+  check-mongo:
+    strategy:
+      matrix:
+        include:
+          - name: Mongo 4.4.2, ReplicaSet, WiredTiger
+            MONGODB_VERSION: 4.4.2
+            MONGODB_TOPOLOGY: replicaset
+            MONGODB_STORAGE_ENGINE: wiredTiger
+            NODE_VERSION: 10
+          - name: Mongo 3.6.21
+            MONGODB_VERSION: 3.6.21
+            NODE_VERSION: 10
+          - name: Redis Cache
+            PARSE_SERVER_TEST_CACHE: redis
+            NODE_VERSION: 10
+          - name: Node 12.12.0
+            NODE_VERSION: 12.12.0
+          - name: Node 14.15.1
+            NODE_VERSION: 14.15.1
+    name: ${{ matrix.name }}
+    timeout-minutes: 30
+    runs-on: ubuntu-20.04
+    services:
+      redis:
+        image: redis
+        ports:
+            - 6379:6379
+    env:      
+      MONGODB_VERSION: ${{ matrix.MONGODB_VERSION }}
+      MONGODB_TOPOLOGY: ${{ matrix.MONGODB_TOPOLOGY }}
+      MONGODB_STORAGE_ENGINE: ${{ matrix.MONGODB_STORAGE_ENGINE }}
+      PARSE_SERVER_TEST_CACHE: ${{ matrix.PARSE_SERVER_TEST_CACHE }}
+      NODE_VERSION: ${{ matrix.NODE_VERSION }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.NODE_VERSION }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.NODE_VERSION }}
+      - name: Cache Node.js modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.OS }}-node-
+            ${{ runner.OS }}-
```

Comment:
twice runner os use node version 😄

Reply:
matrix.NODE_VERSION

Reply:
Took from the GitHub docs example: https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-nodejs#example-caching-dependencies

But honestly I didn't understand well. How do you suggest we should go?

Reply:
i think this should works

```
- uses: actions/cache@v2
  with:
    path: ~/.npm
    key: ${{ runner.os }}-node-${ matrix.NODE_VERSION}-${{ hashFiles('**/package-lock.json') }}
    restore-keys: |
      ${{ runner.os }}-node-${ matrix.NODE_VERSION}-
```

Reply:
done